src/zope/locking/tokens.py

```diff
@@ -150,21 +148,50 @@ def set(self, value):
             else:
                 if value < NO_DURATION:
                     raise ValueError('duration may not be negative')
-                self._expiration = utils.now() + value
+                self._expiration = self._started + value
```

Comment:
Aaugh, the change in indentation makes the diff really misleading.  I guess it can't be avoided!

Reply:
@mgedmin you can ignore whitespace-changes by appending ?w=1 to the url: https://github.com/zopefoundation/zope.locking/pull/1/files?w=1